core/sql/parser/sqlparser.y

```diff
@@ -6579,6 +6581,9 @@ table_reference : table_name_and_hint
               | rel_subquery_and_as_clause
                 { $$ = $1; }
 
+              | with_clause_list
```

Comment:
Here is the syntax rule from the ANSI standard:

<query expression> ::=
[ <with clause> ] <query expression body>